content/docs/forwarding-refs.md

```diff
@@ -1,76 +1,78 @@
 id: forwarding-refs
-title: Forwarding Refs
+title: Ref'leri yönlendirme
 permalink: docs/forwarding-refs.html
 ---
 
-Ref forwarding is a technique for automatically passing a [ref](/docs/refs-and-the-dom.html) through a component to one of its children. This is typically not necessary for most components in the application. However, it can be useful for some kinds of components, especially in reusable component libraries. The most common scenarios are described below.
+Ref yönlendirme, bir [ref](/docs/refs-and-the-dom.html)'i üst bileşenlerden alt bileşenlerin birine otomatik olarak aktarma tekniğidir. Bu genellikle uygulamadaki çoğu bileşen için gerekli değildir. Ama, bazı bileşen türleri için faydalı olabilir, özellikle yeniden kullanılabilir bileşen kütüphaneleri. En yaygın senaryolar aşağıda açıklanmaktadır.
 
-## Forwarding refs to DOM components {#forwarding-refs-to-dom-components}
+## Ref'leri DOM bileşenlerine aktarmak {#forwarding-refs-to-dom-components}
 
-Consider a `FancyButton` component that renders the native `button` DOM element:
+Yerel (native) `button` öğesisini oluşturan `FancyButton` bileşeni düşünün:
 `embed:forwarding-refs/fancy-button-simple.js`
 
-React components hide their implementation details, including their rendered output. Other components using `FancyButton` **usually will not need to** [obtain a ref](/docs/refs-and-the-dom.html) to the inner `button` DOM element. This is good because it prevents components from relying on each other's DOM structure too much.
+React bileşenleri, uygulama ayrıntılarını gizler. `FancyButton` bileşenini kullanan diğer bileşenler, genellikle, inner `button` için [ref oluşturması](/docs/refs-and-the-dom.html) **gerekmez**. Bu iyi bir şeydir, çünkü bileşenlerin birbirilerinin DOM yapısına fazla bağımlı olmasını önler.
 
-Although such encapsulation is desirable for application-level components like `FeedStory` or `Comment`, it can be inconvenient for highly reusable "leaf" components like `FancyButton` or `MyTextInput`. These components tend to be used throughout the application in a similar manner as a regular DOM `button` and `input`, and accessing their DOM nodes may be unavoidable for managing focus, selection, or animations.
+Her ne kadar bu kapsülleme (encapsulation) `FeedStory` veya `Comment` gibi uygulama seviyesi bileşenler için arzu edilirse de, `FancyButton` veya `MyTextInput` gibi yüksek oranda yeniden kullanılabilir "yaprak" bileşenler için sakıncalı olabilir. Bu bileşenler uygulama boyunca normal bir DOM `button` ve `input` öğeleri gibi benzer şekilde kullanılma eğilimindedir, odaklama, seçim veya animasyonları yönetmek için DOM node'larına erişmek kaçınılmaz olabilir.
 
-**Ref forwarding is an opt-in feature that lets some components take a `ref` they receive, and pass it further down (in other words, "forward" it) to a child.**
+**Ref yönlendirme, bazı bileşenlerin aldıkları bir ref'yi almasını ve daha alt elemene aktarmasını sağlayan bir etkinleştirme özelliğidir**
```

Comment:
`ref'yi` -> `ref'i`

There are several places in this file that this change should be applied to. Please check those places, too.